pkg/xds/sync/dataplane_proxy_builder.go

```diff
@@ -136,6 +141,28 @@ func (p *DataplaneProxyBuilder) resolveVIPOutbounds(meshContext xds_context.Mesh
 					continue
 				}
 			}
+		} else {
+			if len(reachableServices) != 0 && reachableBackends == nil {
```

Comment:
add explanation in the comment. That we don't want to ruin performance of existing deployments that has reachable services defined already